packages/logs/src/domain/trackNetworkError.ts

```diff
@@ -46,6 +51,38 @@ export function trackNetworkError(configuration: LogsConfiguration, errorObserva
   }
 }
 
+// TODO: ideally, computeResponseData should always call the `callback` with a string instead of
+// `any`. But to keep retrocompatibility, in the case of XHR with a `responseType` different than
+// "text", the response data should be whatever `xhr.response` is. This is a bit confusing as Logs
+// event 'stack' is expected to be a string. This should be changed in a future major version as it
+// could be a breaking change.
+export function computeResponseData(
+  { xhr, response, error }: { xhr?: XMLHttpRequest; response?: Response; error?: unknown },
+  configuration: LogsConfiguration,
+  callback: (responseData?: any) => void
+) {
+  if (xhr) {
+    if (typeof xhr.response === 'string') {
+      callback(truncateResponseText(xhr.response, configuration))
+    } else {
+      callback(xhr.response)
+    }
+  } else if (response) {
+    response
+      .clone()
+      .text()
+      .then(
+        monitor((text) => callback(text)),
+        monitor((error) => callback(`Unable to retrieve response: ${error as string}`))
+      )
+  } else if (error) {
```

Comment:
I got a bit confused at first on identifying what `response` and `error` cases are compared to `xhr`. As far as I understand `response` and `error` cases are only on fetch requests. 
Maybe better distinguishing when handling `xhr` vs `fetch` could help.

Reply:
I split the function in three with more explicit namings